scripts/linters/pre_commit_linter.py

```diff
@@ -0,0 +1,528 @@
+list of lint errors to the terminal. If the directory path is passed,
+it will lint all Python and JavaScript files in that directory; otherwise,
+it will only lint files that have been touched in this commit.
+
+This script ignores all filepaths contained within .eslintignore.
+
+=====================
+CUSTOMIZATION OPTIONS
+=====================
+1.  To lint only files that have been touched in this commit
+        python -m scripts.linters.pre_commit_linter
+
+2.  To lint all files in the folder or to lint just a specific file
+        python -m scripts.linters.pre_commit_linter --path filepath
+
+3.  To lint a specific list of files (*.js/*.py only). Separate files by spaces
+        python -m scripts.linters.pre_commit_linter
+            --files file_1 file_2 ... file_n
+
+4.  To lint files in verbose mode
+        python -m scripts.linters.pre_commit_linter --verbose
+
+5. To lint a specific list of file extensions (*.js/*.py only). Separate file
+    extensions by spaces
+        python -m scripts.linters.pre_commit_linter
+            --only-check-file-extensions .py .js
+
+Note that the root folder MUST be named 'oppia'.
+ """
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import argparse
+import fnmatch
+import multiprocessing
+import os
+import subprocess
+import sys
+import threading
+
+import python_utils
+
+# Install third party dependencies before proceeding.
+from . import codeowner_linter
+from . import css_linter
+from . import general_purpose_linter
+from . import html_linter
+from . import js_ts_linter
+from . import python_linter
+from .. import concurrent_task_utils
+from .. import install_third_party_libs
+
+install_third_party_libs.main()
+
+_PARSER = argparse.ArgumentParser()
+_EXCLUSIVE_GROUP = _PARSER.add_mutually_exclusive_group()
+_EXCLUSIVE_GROUP.add_argument(
+    '--path',
+    help='path to the directory with files to be linted',
+    action='store')
+_EXCLUSIVE_GROUP.add_argument(
+    '--files',
+    nargs='+',
+    help='specific files to be linted. Space separated list',
+    action='store')
+_PARSER.add_argument(
+    '--verbose',
+    help='verbose mode. All details will be printed.',
+    action='store_true')
+_EXCLUSIVE_GROUP.add_argument(
+    '--only-check-file-extensions',
+    nargs='+',
+    help='specific file extensions to be linted. Space separated list',
+    action='store')
+
+# NOTE TO DEVELOPERS: This should match the version of Node used in common.py.
+NODE_DIR = os.path.abspath(
+    os.path.join(os.getcwd(), os.pardir, 'oppia_tools', 'node-10.18.0'))
+
+if not os.getcwd().endswith('oppia'):
+    python_utils.PRINT('')
+    python_utils.PRINT(
+        'ERROR    Please run this script from the oppia root directory.')
+
+_PARENT_DIR = os.path.abspath(os.path.join(os.getcwd(), os.pardir))
+_PYLINT_PATH = os.path.join(_PARENT_DIR, 'oppia_tools', 'pylint-1.9.4')
```

Comment:
Please define this version in `common.py` and do make sure references like the ones here: 
https://github.com/oppia/oppia/blob/develop/scripts/install_third_party_libs.py#L56 are using the version in `common.py`. https://github.com/oppia/oppia/blob/develop/scripts/install_third_party_libs.py#L161 will also need to be updated.

Reply:
Done.